### PR TITLE
feat: allow users to delete wip submissions

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves this [comment](https://www.notion.so/artsy/Ability-to-delete-a-WIP-submission-31cda3b563ec4d8aa003a4bc46f44fb3?pvs=4)

### Description

This PR allows users to delete WIP submissions using long press.

https://github.com/artsy/eigen/assets/11945712/8773139d-7851-4ff3-96fa-f9ef6d4ccee2

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- allow users to delete wip submissions - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/SellWithArtsy/Components/Header.tsx [hunk 1]
```diff
@@ -49,8 +49,27 @@ export const Header: React.FC<HeaderProps> = (props) => {
...
+                        GlobalStore.actions.artworkSubmission.setDraft(null)
+                      },
+                      style: "destructive",
+                    },
+                  ]
+                )
+              }}
             >
-              <Flex py={1} flexDirection="column">
+              <Flex pb={1} flexDirection="column">
```
Comment (MounirDhahri) [inline src/app/Scenes/SellWithArtsy/Components/Header.tsx:72]: no need for `pt` here

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (allow users to delete wip submissions - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 19146fd6b8970deab97a5907f08e9c02767fe26b
</p>

**Review by olerichter00 — APPROVED** [review]

**Review by olerichter00 — COMMENTED** [review]
I've probably deleted it when updating the design

**Comment by MounirDhahri** [comment]
no worries, glad we got it back

**Comment by anandaroop** [comment]
🙌🏽

**Pull request merged**